src/instructlab/model/download.py

```diff
@@ -12,20 +16,250 @@
+        self.download_dest = download_dest
+
+    def _build_oci_model_file_map(self, oci_model_path: str) -> dict:
+        """
+        Helper function to build a mapping between blob files and what they represent
+        """
+        index_hash = ""
+        try:
+            with open(f"{oci_model_path}/index.json", mode="r", encoding="UTF-8") as f:
+                index_ref = json.load(f)
+            match = re.search("sha256:(.*)", index_ref["manifests"][0]["digest"])
+
+            if match:
+                index_hash = match.group(1)
+            else:
+                click.echo(f"could not find hash for index file at: {oci_model_path}")
+                raise click.exceptions.Exit(1)
+        except FileNotFoundError as exc:
+            raise ValueError(f"file not found: {oci_model_path}/index.json") from exc
+        except json.JSONDecodeError as exc:
+            raise ValueError(
+                f"could not read JSON file: {oci_model_path}/index.json"
+            ) from exc
+        except Exception as exc:
+            raise ValueError("unexpected error occurred: {e}") from exc
+
+        try:
+            with open(
+                f"{oci_model_path}/blobs/sha256/{index_hash}",
+                mode="r",
+                encoding="UTF-8",
+            ) as f:
+                index = json.load(f)
+        except FileNotFoundError as exc:
+            raise ValueError(f"file not found: {oci_model_path}/index.json") from exc
+        except json.JSONDecodeError as exc:
+            raise ValueError(
+                f"could not read JSON file: {oci_model_path}/index.json"
+            ) from exc
+        except Exception as exc:
+            raise ValueError("unexpected error occurred: {e}") from exc
+
+        title_ref = "org.opencontainers.image.title"
+        oci_model_file_map = {}
+
+        for layer in index["layers"]:
+            match = re.search("sha256:(.*)", layer["digest"])
+
+            if match:
+                blob_name = match.group(1)
+                oci_model_file_map[blob_name] = layer["annotations"][title_ref]
+
+        return oci_model_file_map
+
+    def download(self):
+        click.echo(
+            f"Downloading model from OCI registry: {self.repository}@{self.release} to {self.download_dest}..."
+        )
+
+        os.makedirs(self.download_dest, exist_ok=True)
+        model_name = self.repository.split("/")[-1]
+        oci_dir = f"{DEFAULTS.OCI_DIR}/{model_name}"
+        os.makedirs(oci_dir, exist_ok=True)
+
+        try:
+            subprocess.run(
+                [
+                    "skopeo",
+                    "copy",
+                    f"{self.repository}:{self.release}",
+                    f"oci:{oci_dir}",
+                ],
+                check=True,
+            )
+        except FileNotFoundError as e:
+            raise FileNotFoundError(
+                "skopeo not detected, but required to perform downloads from OCI registries. Exiting",
```

Comment:
```suggestion
                "skopeo not installed, but required to perform downloads from OCI registries. Exiting",
```

Also, if you don't e` please remove it.

Reply:
done, thanks